src/core/border_router/routing_manager.hpp

```diff
@@ -632,13 +633,16 @@ class RoutingManager : public InstanceLocator
             kDeprecating,
         };
 
+        void PublishAndAdvertise(void);
+        void Deprecate(void);
         void EnterAdvertisingState(void);
         void AppendCurPrefix(Ip6::Nd::RouterAdvertMessage &aRaMessage);
         void AppendOldPrefix(Ip6::Nd::RouterAdvertMessage &aRaMessage);
 
-        using ExpireTimer = TimerMilliIn<RoutingManager, &RoutingManager::HandleLocalOnLinkPrefixTimer>;
+        using ExpireTimer = TimerMilliIn<RoutingManager, &RoutingManager::HandleOnLinkPrefixManagerTimer>;
 
-        Ip6::Prefix mPrefix;
+        Ip6::Prefix mLocalPrefix;
+        Ip6::Prefix mDiscoveredPrefix;
```

Comment:
What about preserving the name "Favored Discovered Prefix"? The name worked for me :)
```suggestion
        Ip6::Prefix mFavoredDiscoveredPrefix;
```

Reply:
To be honest, I also like "favored discovered prefix" name :)...  
I know that I tend to use long names so sometimes try to actively edit myself (force myself to useshorter names)- Changed it back now to `mFavoredDiscoveredPrefix`.

Reply:
Thanks!